### ISSUE TITLE
Setting to opt out of quote posts or quote boosts

### ISSUE THREAD

**Issue by brendanjones** [mastodon/mastodon#25735]
### The aim of this issue is to define 

1. A universal post attribute that communicates if the author of a post *wants* their post to be quote-posted or not, and
2. Any UI elements required for the user to control that post attribute.

### Implementation

1. Add a setting to `/settings/preferences/other` to "Allow your posts to be quoted by default." This setting would be off by default.
2. Add an attribute to posts to define whether it's quotable or not. *I'm not a dev but I imagine something like add a boolean (true/false) field `quotes`.*
3. Add a toggle to the post composer UI, so users can turn quoting on/off per post.

### Motivation

Quote posting is already here on the Fediverse and Mastodon. Calckey has it, various apps have it. Technically speaking there's no way to stop quote posting. However if there is a per-post attribute that signals the post author's desires about quote posting for that post, then if servers or clients don't respect that attribute they can be defederated and/or avoided.

Since quote posting is already is already here, I'd like to see this post attribute added already and not wait until quote posting is built into Mastodon natively.

This issue is NOT for the implementation of quote post features. There are already a bazillion issues for that: https://github.com/mastodon/mastodon/issues/309, https://github.com/mastodon/mastodon/issues/14597, https://github.com/mastodon/mastodon/issues/19889, https://github.com/mastodon/mastodon/issues/22793 are the ones I know of. 

I looked for a duplicate issue but can't find one - only discussions about this in issues for implementation of quote posts.

**Comment by renchap** [comment]
As you mentioned, Quote Posts does not exists in Mastodon. There is no standard defining them.

We want to implement this feature, but this will require changes to the ActivityPub protocol to, in part, allow the feature you are asking for and give authors a way to control who can quote their posts. This is currently the major blocker for the feature.

There are some proposals around this, but none have been approved yet. This is a pre-requisite to any implementation of this feature (and it will most probably ship at the same time as Quote Posts)

**Comment by brendanjones** [comment]
Why wait to ship with the quote posts feature? Other fediverse software and Mastodon apps (e.g. Ice Cubes) already have quote posts, so the sooner the controls are released, the better.

**Comment by renchap** [comment]
Because we do not want to add our own Quote Post implementation (as some other software did) without also having it properly handled by the protocol, with some privacy controls, like the ability to prevent people to quote your posts.

**Comment by trwnh** [comment]
Define "quote post", though? Because the following are mutually exclusive:

> Other fediverse software and Mastodon apps (e.g. Ice Cubes) already have quote posts

> (URL sharing is not quote posting)

What these other softwares do is basically just share the URL in a new post, then if they encounter any URL in a post they will try to fetch it as a post, and if successful then that post gets embedded as a preview. If "URL sharing is not quote posting", then what makes it a "quote post"?

**Comment by brendanjones** [comment]
> Define "quote post", though?

Anything that looks like a quote post to a user, such as most people's point of reference for quote posts: Twitter quote tweets. Also, anything that gets called a quote post. Yes I realise that's fairly arbitrary. You're saying what's being done already in these apps is not quote posts because of the mechanic that produces them, but I and most users are unaware of the exact mechanic behind how the deed is done. We get told it's a quote post, and it looks like one? It's a quote post.

**Comment by brendanjones** [comment]
> with some privacy controls, like the ability to prevent people to quote your posts.

Yeah and what I'm saying is we don't need to wait to know the *exact* protocol structure/semantics in order to already develop those privacy controls - implementation items 1,3 & 4 in my list can already be done by anybody without waiting on the AP spec. 

So okay, I maybe went too broad with my ticket scope and "2, Add an attribute to posts to define whether they're quotable or not." should be put into a separate ticket. But it still makes sense to me to track quote post implementation and quote post controls in separate tickets.

**Comment by trwnh** [comment]
I'm not sure the target should be what Twitter does or did, because that's what we're historically not interested in doing a carbon-copy of. Consequently, it makes sense to try to define a "quote post" in more concrete terms, because how you choose to render a link is entirely up to the client. You could even turn ordinary replies into such "quote posts" if you just rendered the immediate parent as a preview above the current post. Is this objectionable enough that it warrants a preference for preventing it? What are you preventing -- the interpretation of a certain client?

So there's a few things that make up a "quote", such as:

- What is the link relation between the quote and the quoted? Is it a response like `inReplyTo`, or something else?
- Is the linked resource shown as a preview? Is the preview shown above, below, or in the middle of the post?
- Can you quote multiple objects or fragments? Why or why not? How?
- Should it be included in the previous thread? (Although Mastodon does not currently have a first-class concept of a "thread".)

Which is the objectionable part? What are we allowing people to express a preference to opt in or out of? Compare to the `discoverable` toggle, which allows profiles to be marked `discoverable` and opt into "discovery features" -- being included in profile directories which can be searched through, for example. A similar preference that makes sense to me would be perhaps `embeddable` or `reproducible` or something like that, expressing a preference to not have your post be reproduced in an embed or preview or iframe or what have you. See #4883 for this. But the scope should be clearly defined. One imagines such a preference might imply always fetching such a post from the origin, hiding it behind a clickthrough: "The author of this post has expressed that they do not want this post to be reproduced or embedded anywhere else. Click here to [view it at / load it from] the origin."

**Comment by samwightt** [comment]
Why would this be off by default and not on? That's not consistent with any other feature in Mastodon. Any user can follow you until you turn follow requests on. Any user can view your boosts by default until you change the default visibility. Consistently the default behavior in Mastodon is to *encourage* new types of interaction by default, not restrict them.

Personally if this is turned off by default, the third-party tools I develop (and other folks I know developing them) will likely not respect it because it kinda becomes useless. Most users would expect something like this to be *on* by default and wouldn't think to check settings to enable it.

**Comment by samwightt** [comment]
Additionally I fail to see why this needs to be a setting or a flag in ActivityPub somewhere when there's no enforcement mechanism outside of defederation. Post visibility and follow requests are self-enforcing because the user has control over them. Users do not have control over quote-posting because a quote post (as it is implemented by all other apps right now) is just a link to their post. You cannot control whether people can link to your post or not, only whether they can see it.

I struggle to see a reason why other Fediverse apps would take the time out of their busy development lifecycle to build very specific support for this very specific Mastodon feature when Mastodon a) has consistently refused to even come to the table on implementing a quote post feature, and b) there is nothing self-enforcing in the spec for it.

**Comment by neffscape** [comment]
Sorry, but I can't understand what the issue is with this function. It seems to me that Mastodon already has all the necessary tools to interpret the author's intent to allow 'quote boosts' in the post privacy settings: if a post is public, the author, in my opinion, exposes it to the possibility of being shared, reposted, and thus commented on. Mastodon already sends notifications for boosts; extending these notifications to 'quote boosts,' which, as I understand, are posts containing a URL to another post, should suffice. I also believe that the risk of 'quote boosts' being used maliciously is significantly reduced in an environment like Mastodon. Instance administrators, when establishing their rules, can clearly state how they want 'quote boosts' to be used. If they find a user violating these rules, they can take appropriate actions, just as they would with other non-compliant or unacceptable content according to the instance's guidelines. 
In conclusion - and I say this with all the love and respect I have for Eugen, the development team, and the project - I believe the problem is overestimated, and further delaying the implementation of this function could hinder the migration of news outlets and many users who are currently seeking an alternative to X, impeding the adoption of Mastodon. It would be a significant loss and a missed opportunity for Mastodon, user freedom, and the entire fediverse.

**Comment by Packbat** [comment]
Just going off my experience on a server without quote-posting, the fallback behavior of the 'quoted' post just showing up as a link seems ... fine? I don't see any real harm in making quote-posts a opt-in feature, either on a post-by-post or account-by-account level.

...but also I don't have any meaningful experience with quote posting as a phenomenon. I'd be really curious what folks who do have this experience think of the feature.

**Comment by WisTex** [comment]
Have you ever seen the screenshots of Twitter or X that were then shared on Facebook or Instagram or Mastodon or elsewhere? 

People can quote you simply by taking a screenshot or using their phone to take a picture of their computer's screen. Or they can use an app that does not honor (or even know about) the "do not quote" flag.

I could see the usefulness of this being an advisory flag that Mastodon and other projects _might_ honor, similar to the "Do No Track" flag your browser can send to websites.

But that is not going to stop malicious people from quoting you.

Even if you implement this feature, which I don't think other platforms would adopt, it won't have the effect you intend.

**Comment by shaedrich** [comment]
> Why would this be off by default and not on? […] Personally if this is turned off by default, the third-party tools I develop (and other folks I know developing them) will likely not respect it because it kinda becomes useless. Most users would expect something like this to be on by default and wouldn't think to check settings to enable it.

@samwightt At least in Europe with GDPR and the like, opt-in _is_ the desired state of "privacy by default". However, your concerns that people might _forget_ to opt-in, is of course justified. I faced that issue myself with some software, I'm developing, but I'm convinced that this _shouldn't_ be an argument _against_ privacy but _in favor_ of clear and supportive UI and UX. Meaning, to have "informed consent" the user has to get a little tour when they start on Mastodon where they are asked once(!) to turn this feature on and when they decide against it, they are not bothered again.

> I could see the usefulness of this being an advisory flag that Mastodon and other projects might honor, similar to the "Do No Track" flag your browser can send to websites.

@WisTex It seems to have become a sport to find the most far-fetched lie about why it is "technically impossible" to implement DNT to put in everyone's privacy policy these days. So maybe not the best example 😉

**Comment by trwnh** [comment]
i *think* this ended up being defined in https://w3id.org/fep/044f as `https://gotosocial.org/ns#canQuote` (nested under `https://gotosocial.org/ns#interactionPolicy`, but without recognizing the other GTS-specific policies)?

Currently, a user preference is available under `/settings/preferences/posting_defaults`:

<img width="1182" height="890" alt="Image" src="https://github.com/user-attachments/assets/af5b0fe9-f0aa-4b97-8bfe-2d486442b674" />
